src/librustc_mir/transform/mod.rs

```diff
@@ -267,54 +269,78 @@ fn run_optimization_passes<'tcx>(
     def_id: DefId,
     promoted: Option<Promoted>,
 ) {
+    let post_borrowck_cleanup: &[&dyn MirPass<'tcx>] = &[
+        // Remove all things only needed by analysis
+        &no_landing_pads::NoLandingPads::new(tcx),
+        &simplify_branches::SimplifyBranches::new("initial"),
+        &remove_noop_landing_pads::RemoveNoopLandingPads,
+        &cleanup_post_borrowck::CleanupNonCodegenStatements,
+        &simplify::SimplifyCfg::new("early-opt"),
+        // These next passes must be executed together
+        &add_call_guards::CriticalCallEdges,
+        &elaborate_drops::ElaborateDrops,
+        &no_landing_pads::NoLandingPads::new(tcx),
+        // AddMovesForPackedDrops needs to run after drop
+        // elaboration.
+        &add_moves_for_packed_drops::AddMovesForPackedDrops,
+        // `AddRetag` needs to run after `ElaborateDrops`. Otherwise it should run fairly late,
+        // but before optimizations begin.
+        &add_retag::AddRetag,
+        &simplify::SimplifyCfg::new("elaborate-drops"),
+        // No lifetime analysis based on borrowing can be done from here on out.
+    ];
+
+    let optimizations: &[&dyn MirPass<'tcx>] = &[
+        &unreachable_prop::UnreachablePropagation,
+        &uninhabited_enum_branching::UninhabitedEnumBranching,
+        &simplify::SimplifyCfg::new("after-uninhabited-enum-branching"),
+        &inline::Inline,
+        // Lowering generator control-flow and variables has to happen before we do anything else
+        // to them.
+        &generator::StateTransform,
+        &instcombine::InstCombine,
+        &const_prop::ConstProp,
+        &simplify_branches::SimplifyBranches::new("after-const-prop"),
+        // Run deaggregation here because:
+        //   1. Some codegen backends require it
+        //   2. It creates additional possibilities for some MIR optimizations to trigger
```

Comment:
```suggestion
        //   2. It creates additional possibilities for some MIR optimizations to trigger
        // FIXME(#70073): why is this done here and not in one of the cleanup blocks?
```
I don't think this question has been answered, so let's put it into the code as well.

Reply:
At the very least, this allows `SimplifyLocals` to remove more things [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=175335edafaa908b8d9020f663f89ffd). So it at least has to be run before `SimplifyLocals`.

Reply:
Sure, but that doesn't explain why it's not in `post_borrowck_cleanup`.

I propose to just punt on that issue now -- it is pre-existing and orthogonal to your PR. We should just note it as an open question.

Reply:
If you feel that resolves your question, I can add that to the comment, otherwise I'll add your FIXME

Reply:
What resolves my question? I haven't seen an explanation yet for why this is not in `post_borrowck_cleanup`, and indeed [probably that is just an accident](https://github.com/rust-lang/rust/pull/70073#discussion_r404424193).

Reply:
Sorry, I kept reading that as `pre_codegen_cleanup`. You're totally right.